### PR TITLE
Add URL parameter to drop a cookie

### PR DESCRIPTION
**Pull request by adgad**
/cc @commuterjoy 

Adds a URL parameter that sets specified flags when opting in. This is to enable, as an example, a group of remote testers to access the new Homepage layout before it is live.

Restricted to a manual whitelist of flags, so nobody can do anything malicious!

### CODE REVIEW

File: server/controllers/opt-in.js [hunk 1]
```diff
@@ -9,6 +9,9 @@ const debug = require('debug')('ft-next-opt-in-out-debug');
 const logger = require('ft-next-express').logger;
 const doOptIn = require('../lib/opt-in');
 
+
+const publicFlagsWhitelist = ['frontPageLayoutPrototype'];
```
Comment (commuterjoy) [inline server/controllers/opt-in.js]: or I suppose we could have a property as part of flags - allowPublicUse ?

File: server/controllers/opt-in.js [hunk 2]
```diff
@@ -48,13 +51,26 @@ function getSessionDetails(req){
 	});
 }
 
-function optIn(req, res, referrerUrlObj, optedVia, cookie, newUrl, isAnon){
+function setFlags(res, flagOverrides) {
+	let flags = flagOverrides.split(';');
+	flags = flags.filter(flag => (publicFlagsWhitelist.indexOf(flag.split(':')[0]) >= 0));
+	flags = flags && flags.length ? flags.join(';') : null;
+	if(flags) {
+		res.cookie('next-flags', flags, { maxAge: config.COOKIE_MAX_AGE, domain: 'ft.com' });
```
Comment (commuterjoy) [inline server/controllers/opt-in.js]: could really scope to next.ft.com ?

**Comment by commuterjoy** [comment]
looks good.

File: server/controllers/opt-in.js (hunk 1 above)
Comment (adgad) [inline server/controllers/opt-in.js]: Yup that would be ideal - but probably only worth thinking about if this feature ever becomes more widely used

Comment (commuterjoy) [inline server/controllers/opt-in.js]: fair enough. yeah, ideally flags would not be a public thing.

**Pull request merged**